hathor/consensus/consensus.py

```diff
@@ -210,6 +210,7 @@ def _remove_transactions(
         for tx in txs:
             tx_meta = tx.get_metadata()
             assert not tx_meta.validation.is_checkpoint()
+            assert bool(tx_meta.voided_by), 'removed txs must be voided'
```

Comment:
Why?

Reply:
This is an invariant that we currently uphold in our code and if we ever introduce something that fails this assertion, we must know.

The `_remove_transactions()` method is called with the results from `storage.compute_transactions_that_became_invalid()`, which calls `tx.set_validation(ValidationState.INVALID)` for each tx to be removed. This setter calls `_mark_partially_validated()` which sets `b'pending-validation'` in the `voided_by`.

Reply:
And most importantly, semantically it makes sense: we only ever remove transactions that became invalid, and by definition if transactions are invalid they must have also be voided.

Reply:
It was not that clear to me that invalid transactions must be voided. I guess the consensus is the only part of the code that can void a transaction while the transaction can get invalid for other reasons.

I feel that this assert will fail when a transaction fails the reward lock verification after a reorg. In this case, `TransactionStorage.compute_transactions_that_became_invalid()` will set the validation to `INVALID` but it will not necessarily be voided.

@jansegre What do you think?

Reply:
After a quick conversation, I saw that the `set_validation()` method will add a marker to the `meta.voided_by`. Thanks!